### PR TITLE
1030 - Tasks in dc.data.DiskDataset.merge

### PR DESCRIPTION
**Pull request by fmonta**
This PR is a fix for #1030, however that fix assumes that the DiskDatasets to merge have the same number of tasks. This works for my use case but not sure about the general contract of that method.
I also added a unittest for merge().

### CODE REVIEW

File: deepchem/data/datasets.py [hunk 1]
```diff
@@ -920,7 +920,7 @@ def generator():
         X, y, w, ids = (dataset.X, dataset.y, dataset.w, dataset.ids)
         yield (X, y, w, ids)
```
Comment (lilleswing) [inline deepchem/data/datasets.py]: We need to do a check here that we are passing in only DiskDatasets.

```python
tasks = []
for dataset in datasets:
  if isintance(dataset, deepchem.data.DiskDataset):
    tasks = dataset.tasks
    break
return DiskDataset.create_dataset(generator(), data_dir=merge_dir, tasks=tasks)
```

**Comment by lilleswing** [comment]
Great catch!  after the type check this should be good to go!

**Comment by lilleswing** [comment]
just yapf left
```python
pip install yapf==0.20.0
yapf -i deepchem/data/datasets.py
deepchem/data/tests/test_datasets.py
```

**Comment by fmonta** [comment]
done

**Comment by lilleswing** [comment]
LGTM

**Pull request merged**